theano/gof/opt.py

```diff
@@ -533,19 +544,64 @@ def process_node(self, fgraph, node):
             if len(node.inputs) != len(candidate.inputs):
                 continue
 
+            cand_has_assert = False
+
+            # Get input list of the candidate with assert removed
+            cand_inputs_assert_removed = []
+            for i in candidate.inputs:
+                if i.owner and isinstance(i.owner.op,
+                                          theano.tensor.opt.Assert):
+                    cand_has_assert = True
+                    cand_inputs_assert_removed.append(i.owner.inputs[0])
+                else:
+                    cand_inputs_assert_removed.append(i)
+
+            # if the current node has assert input, it should not be
+            # replaced with a candidate node which has no assert input
```

Comment:
I thought that in every iteration on `node`, only `node` was the one that should be replaced. So it is ok to replace the `candidate`?